### PR TITLE
Build/dependency cleanup

### PR DESCRIPTION
**Pull request by adam-mesos**
- Fix log4j exception when starting scheduler
- Exit build-hdfs early if compile fails
- Copy required dependency directories, rather than deleting cruft
- Remove unnecessary sleeps

### CODE REVIEW

File: bin/hdfs-mesos-namenode [hunk 1]
```diff
@@ -18,23 +18,17 @@ fi
 trap "{ $DIR/mesos-killtree "$$" ; exit 0; }" EXIT
 
 function bootstrap_standby() {
-  echo "Pausing for 10s before bootstrapping standby namenode"
```
Comment (elingg) [inline bin/hdfs-mesos-namenode]: I'm fine with getting rid of the sleep commands, but this needs to be consistent for all scripts so the other scripts also need to be modified.

File: bin/build-hdfs [hunk 2]
```diff
@@ -61,21 +51,35 @@ else
...
+# Create dist
 if [ ! -d $DIST ]; then
     echo "Creating new $DIST dist folder"
     mkdir -p $DIST
-    cp -R $HADOOP_DIR/* $DIST
 else
     echo "($DIST already exists, skipping create)"
 fi
 
+# Copy to dist
```
Comment (elingg) [inline bin/build-hdfs]: glad you are copying only the necessary dirs!

File: bin/hdfs-mesos-namenode (hunk 1 above)
Comment (adam-mesos) [inline bin/hdfs-mesos-namenode]: The other sleeps seem necessary. Within `hdfs-mesos-namenode/journalnode/zkfc`, the `sleep 10` inside the `while [ true ]` loop guarantees things have settled down after the process exited before restarting it. The sleep in `mesos-killtree` gives the processes being killed time for a graceful shutdown between `SIGTERM` and `SIGKILL`.
So these are the only ones slowing down namenode (cluster) startup unnecessarily.

**Comment by adam-mesos** [comment]
@elingg PTAL, ready to merge?

File: bin/hdfs-mesos-namenode (hunk 1 above)
Comment (elingg) [inline bin/hdfs-mesos-namenode]: Fair enough!

**Pull request merged**